pkg/route/api/validation/validation.go

```diff
@@ -46,6 +46,24 @@ func ValidateRoute(route *routeapi.Route) field.ErrorList {
 	if route.Spec.To.Kind != "Service" {
 		result = append(result, field.Invalid(specPath.Child("to", "kind"), route.Spec.To.Kind, "must reference a Service"))
 	}
+	if route.Spec.To.Weight != nil && (*route.Spec.To.Weight < 1 || *route.Spec.To.Weight > 256) {
+		result = append(result, field.Invalid(specPath.Child("to", "weight"), route.Spec.To.Weight, "weight must be an integer between 1 and 256"))
+	}
+
+	if len(route.Spec.AlternateBackends) > 3 {
+		result = append(result, field.Required(specPath.Child("to"), "cannot specify more than 3 additional services"))
+	}
+	for _, svc := range route.Spec.AlternateBackends {
+		if len(svc.Name) == 0 {
+			result = append(result, field.Required(specPath.Child("to", "name"), ""))
```

Comment:
This is wrong, should be `specPath.Child("alternateBackends").Index(i).Child("name")`